shared/src/main/scala/com/comcast/ip4s/Cidr.scala

```diff
@@ -118,6 +133,23 @@ final class Cidr[+A <: IpAddress] private (val address: A, val prefixBits: Int)
 
 object Cidr {
 
+  /** A normalized cidr range, of which the address is identical to the prefix.
+    *
+    * This means the address will never have any bits set outside the prefix. For example, a range
+    * such as 192.168.0.1/31 is not allowed.
+    */
+  final class Strict[+A <: IpAddress] private (override val prefix: A, override val prefixBits: Int)
```

Comment:
Nit: can't these be passed along as ctor args?

```suggestion
  final class Strict[+A <: IpAddress] private (prefix: A, prefixBits: Int)
```

Reply:
for the prefix bits, yes, that's just plain better.

For prefix, if it's an override val, it overrides `def prefix: A =
    address.transform(_.masked(Ipv4Address.mask(prefixBits)), _.masked(Ipv6Address.mask(prefixBits)))` with the val, which adds a field but potentially saves a bit of useless work. I guess I'd rather take the extra field?

Reply:
@armanbilge what do you think?

Reply:
@martijnhoekstra sorry, I missed this.

Instead of adding an extra redundant field, can't we just do `override def prefix = address`?

Reply:
- https://github.com/Comcast/ip4s/pull/545